src/index.ts

```diff
@@ -228,7 +230,7 @@ export class GraphQLClient {
       query: rawRequestOptions.query,
       variables: rawRequestOptions.variables,
       headers: {
-        ...resolveHeaders(headers),
+        ...resolveHeaders(callOrIdentity(headers)),
```

Comment:
Note that dynamic headers were added only to GraphQLClient methods - the request and rawRequest argument `requestHeaders` doesn't accept a function